Carmen Cortez_Assignment 4_PCA.Rmd

```diff
@@ -0,0 +1,154 @@
+```
+
+## Think about which components you would drop and make a decision
+
+## Part II
+
+```{r}
+#Now, create a data frame of the transformed data from your pca.
+
+D3 <- as.data.frame(pca$x)
+
+#Attach the variable "mean_correct" from your original data frame to D3.
+
+D4 <- cbind(D3, as.data.frame(D1$mean_correct))
+
+#Now re-run your scatterplots and correlations between the transformed data and mean_correct. If you had dropped some components would you have lost important infomation about mean_correct?
+
+COR2 <- cor(D4)
+
+corrplot(COR2, order="AOE", method="circle", tl.pos="lt", type="upper",        
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
+```
+## Now print out the eigenvectors (often called loadings) for the components you generated:
+
+```{r}
+pca$rotation
+#Examine the eigenvectors, notice that they are a little difficult to interpret. It is much easier to make sense of them if we make them proportional within each component
+
+loadings <- abs(pca$rotation) 
+#abs() will make all eigenvectors positive
+
+sweep(loadings, 2, colSums(loadings), "/") 
+#sweep() computes each row as a proportion of the column. (There must be a way to do this with dplyr()?)
+
+#Now examine your components and try to come up with substantive descriptions of what some might represent? The correlations between the principal components and the original variables are copied into the table below. The first principal componet is strongly correctlated with two of the original values. The first component increases with increasing mean_hint and mean_attempt scores. It would follow that students with higher number of hints received would perform better. The second principal component increases with only one of the values, increasing prior_percent_correct. This component can be viewed as a measure of how well student perform when the percentage of problems a student has answered correctly prior to this session is higher. The third component decreases with decreasing mean_confidence. This suggests that performance decreases when average confidence each student has in their ability to answer the problems decreases. The fourth principal component increases only with only one of the values, decreasing prior_prob_count. This measure can be viewed as how poorly students fair when the number of problems students have answered in the system prior to this session decreases. The fifth principal component is strongly correlated with two of the original values, problems attempted and means_attempt. It would make sense to see an increase in performance when the average number of correct answers on first attempt increases and the average number of attempts decreases. In contrast the fourth principal, the sixth principal increases with only one of the values, increasing mean_hint. This indicates that students tend to perform better when they have access to additional support.
```

Comment:
Good work with analyzing how the components are related to other variables. It's useful to think of them in terms of what they generally represent. For example, since PC1 is strongly correlated with mean_hints or mean_attempts, maybe it represents some sort of eagerness to click through the questions.